csrc/ec_gen.cpp

```diff
@@ -203,6 +207,9 @@ JNIEXPORT void JNICALL Java_com_amazon_corretto_crypto_provider_EcGen_generateEc
         ex.throw_to_java(pEnv);
     }
 
-    EC_KEY_free(ecParams);
     EC_KEY_free(key);
+    EC_KEY_free(ecParams);
```

Comment:
Please remove the comment related to freeing this object. It's definitely out of date and doesn't make sense to me anymore.